File: Algoritm_Practice/Leetcode/TwoSum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Algoritm_Practice.Leetcode
{
    public static class TwoSum
    {
        public static int[] TwoSumOp(int[] nums, int target)
        {
            HashSet<int> l = new HashSet<int>();
            for (int i = 0; i < nums.Length; i++)
            {
                for (int j = i + 1; j < nums.Length; j++)
                {
                    if (nums[i] + nums[j] == target)
                    {
                        l.Add(nums[i]);
                        l.Add(nums[j]);
                        break;

                    }

                }
            }
            int[] result = l.ToArray();
            return result;
        }
    }
}

File: C++/TwoSum.cpp
#include "TwoSum.h"

std::vector<int> TwoSum::twoSum(std::vector<int>& nums, int target)
{
	std::map<int, int> values;

	for (int i{ 0 }; i < nums.size(); ++i)
	{
		int secondNum = target - nums[i];
		if (values.count(secondNum))
			return { values[secondNum],i };
		values[nums[i]] = i;
	}
	return{};
}
